src/Control/Effect/Cut.hs

```diff
@@ -48,53 +49,64 @@ call m = send (Call m ret)
 
 -- | Commit to the current branch, preventing backtracking within the nearest enclosing 'call' (if any) on failure.
 --
---   prop> run (runNonDet (runCut (cut *> pure a <|> pure b))) == [a, b]
+--   prop> run (runNonDet (runCut (pure a <|> cut *> pure b))) == [a, b]
+--   prop> run (runNonDet (runCut (cut *> pure a <|> pure b))) == [a]
 --   prop> run (runNonDet (runCut (cut *> empty <|> pure a))) == []
 cut :: (Alternative m, Carrier sig m, Member Cut sig) => m ()
 cut = pure () <|> cutfail
 
 
 -- | The result of a nondeterministic branch of a computation.
-data Branch a
-  = Prune
+data Branch m a
+  = Cut
   | None
-  | Some a
-  deriving (Eq, Functor, Ord, Show)
+  | Pure a
```

Comment:
I renamed a couple of the `Branch` constructors to better describe the operations they represent.